domain/src/main/java/com/example/architecture/domain/remote/ArchService.kt

```diff
@@ -0,0 +1,13 @@
+package com.example.architecture.domain.remote
+
+import com.example.architecture.domain.BuildConfig
+import com.example.architecture.domain.remote.response.ArticleListResponse
+import retrofit2.http.GET
+import retrofit2.http.Header
+import retrofit2.http.Headers
+
+interface ArchService {
+    @Headers("Authorization: ${BuildConfig.API_KEY}")
```

Comment:
We should showcase the interceptors here instead of using it like this.

Reply:
Added